packages/express/src/clerkMiddleware.ts

```diff
@@ -0,0 +1,59 @@
+import { createClerkClient } from '@clerk/backend';
+import type { RequestHandler } from 'express';
+
+import { authenticateRequest, setResponseHeaders } from './authenticateRequest';
+import { clerkClient as defaultClerkClient } from './clerkClient';
+import { middlewareNotInvoked } from './errors';
+import type { ClerkMiddleware, ClerkMiddlewareOptions } from './types';
+import { defaultHandler } from './utils';
+
+const usedWithoutInvocation = (args: unknown[]) => {
+  return (
+    args.length === 3 && typeof args[0] === 'object' && typeof args[1] === 'object' && typeof args[2] === 'function'
+  );
+};
+
+const parseHandlerAndOptions = (args: unknown[]) => {
+  return [
+    typeof args[0] === 'function' ? args[0] : undefined,
+    (args.length === 2 ? args[1] : typeof args[0] === 'function' ? {} : args[0]) || {},
+  ] as [RequestHandler | undefined, ClerkMiddlewareOptions];
+};
+
+export const clerkMiddleware: ClerkMiddleware = (...args: unknown[]) => {
+  const [handler, options] = parseHandlerAndOptions(args);
+
+  const shouldInitializeClerkClient = Object.keys(options).length > 0;
+  const clerkClient = shouldInitializeClerkClient ? createClerkClient(options) : defaultClerkClient;
```

Comment:
❓ Why don't we switch to an injection logic here?
Instead of initializing a clerk client, wouldn't it be better to ask the users to inject an instance? That way we can remove the duplicated logic and option props from here.

Reply:
We could do that but with this approach the customers don't need to use the `createClerkClient` (which is a low-level function re-exported from `@clerk/backend`) to avoid using envs for some reason.
What do you think will be the pros / cons of injecting a `clerkClient`?

Reply:
All low-level apis were moved to sub paths of clerk/backend, so I wouldn't consider `createClerkClient` a low-level API right now -  this is the preferred way to use Clerk in CF workers and similar runtimes after all. 

As for injection vs us automatically initializing a client, it might be personal preference, but I think that:
- It's more explicit for the user 
- A method called `clerkMiddleware` should probably not be concerned with creating a clerk client at all. 

The 
```
  const shouldInitializeClerkClient = Object.keys(options).length > 0;
```
makes me think that `clerkMiddlware` does more than it should, as without the named variable I wouldn't be able to understand what `Object.keys(options).length > 0` means

Reply:
I have updated the clerkMiddleware to support `clerkClient` optional instance instead of implicitly creating one.